### PR TITLE
Add example recipe to show how to deploy files into Mender persistent…

### PR DESCRIPTION
**Pull request by drewmoseley**
… data partition

Signed-off-by: Drew Moseley <drew.moseley@mender.io>

### CODE REVIEW

**Comment by drewmoseley** [comment]
@kacf please take a look at this.  This is my initial attempt at deploying files to the persistent data partition using MENDER_DATA_PART_DIR.  Is this consistent with what you were expecting?

File: meta-mender-demo/recipes-mender/hello-mender/hello-mender.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+DESCRIPTION = "Mender test recipe for persistent data files"
+LICENSE = "Apache-2.0"
+
+inherit deploy
+
+do_compile() {
+    dd if=/dev/urandom of=persistent.dat bs=4k count=100
+    dd if=/dev/urandom of=hello-mender bs=4k count=100
```
Comment (kacf) [inline meta-mender-demo/recipes-mender/hello-mender/hello-mender.bb]: Maybe use something predictable so it's easier to see that the right content made it onto the `/data` partition?

**Comment by drewmoseley** [comment]
@kacf I updated this to provide known contents for verification rather than relying on /dev/urandom.

I had not considered what to do about documenting this; it's fairly straightforward as is.  Do we want do put this in the docs given that you are planning a better upstream solution?

**Comment by drewmoseley** [comment]
On further review, we don't want to merge this as is.  Given the bbappend to the core-image-full-cmdline recipe, this will cause any build that doesn't include the hello-mender recipe to fail.  I suppose we could add hello-mender in meta-mender-demo/recipes-core/images/core-image-full-cmdline.bbappend but I'm not sure that's the right approach.  Thoughts?

**Comment by kacf** [comment]
I think that would be the right approach, yes. Why are trying to stick to bug fixes only in pyro; given that this is mostly for demonstration, and doesn't add actual value to the image, I think we should keep it in master only.

**Comment by drewmoseley** [comment]
@kacf I updated this to include the package in the core-image-full-cmdline build from the meta-mender-demo layer.  That should resolve the build failure here and allow this to be merged.

**Comment by drewmoseley** [comment]
Doc updates for this are available here: https://github.com/mendersoftware/mender-docs/pull/162

**Comment by drewmoseley** [comment]
I added the changelog details.

I have no problem with the change you recommend in the mender-sdimg.bbclass.

**Comment by kacf** [comment]
Great, if you can add another commit where you make the chown fix I described, the next test run should succeed.

**Comment by kacf** [comment]
Famous last words :-)

**Comment by drewmoseley** [comment]
chown patch added and tested.

**Comment by kacf** [comment]
Thanks! I'll make sure this lands in pyro as well.

**Comment by kacf** [comment]
Actually, only the second patch, since we decided to keep the demo recipe in master only.

**Pull request merged**